### ISSUE TITLE
AppInfo stores strings incorrectly

### ISSUE THREAD

**Issue by kotauskas** [andybarron/app-dirs-rs#37]
The [`AppInfo`][appinfo] struct stores the `&'static str` type, which fails miserably if any of the fields are borrowed from an owned `String`. It should be changed to use an explicit generic lifetime parameter and store the string slices with that lifetime.

[appinfo]: https://docs.rs/app_dirs/1.2.1/app_dirs/struct.AppInfo.html " "

**Comment by daboross** [comment]
I don't think this wrong, it's just not fully general. I agree that it would be better if it were generic, but the current code won't cause any bugs, and can still function perfectly well with compile-time-known strings.
